### PR TITLE
feat(digits) add error handling and cleanup

### PR DESCRIPTION
**Pull request by masoudDaliriyan**

### CODE REVIEW

File: src/modules/digits/index.ts [hunk 1]
```diff
@@ -8,19 +10,21 @@ const arNums = "٠١٢٣٤٥٦٧٨٩";
...
 		throw new TypeError("the input must be string or number");
 	}
 
-	let string = typeof value === "number" ? String(value) : value;
+	let digitsString = isNumber(digits) ? String(digits) : digits;
 
 	for (let i = 0; i < 10; i++) {
 		const replaceEntoFa = new RegExp("" + i, "g");
-		string = string.replace(replaceEntoFa, faNums[i]);
+		// @ts-ignore
```
Comment (ali-master) [inline src/modules/digits/index.ts:22]: Remove `@ts-ignore` and fix the issue

File: src/modules/digits/index.ts [hunk 2]
```diff
@@ -30,18 +34,18 @@ export function digitsEnToFa(value?: number | string): string {
  *  Persian digits
  *
  */
```
Comment (ali-master) [inline src/modules/digits/index.ts:36]: Please update the functions' JSDocs

File: src/helpers/index.ts [hunk 3]
```diff
@@ -15,3 +16,11 @@ export const replaceArray = (string: string, find: ReplaceArrayDictionary): stri
  * @type regex
  */
 export const ArabicContextualForms = /[ي|ﻱ|ﻲ|ﻚ|ك|ﻚ|ﺔ|ﺓ|ة]/g;
+
+export function isString(value: string): boolean {
```
Comment (ali-master) [inline src/helpers/index.ts:20]: There is no related tests for `isString` and `isNumber` functions.

File: src/helpers/index.ts [hunk 4]
```diff
@@ -15,3 +16,11 @@ export const replaceArray = (string: string, find: ReplaceArrayDictionary): stri
...
+	return {}.toString.call(value) === "[object String]";
```
Comment (ali-master) [inline src/helpers/index.ts:21]: Please create a function:
```js
function getType(value: any): string {
   return Object.prototype.toString.call(value).slice(8, -1).toLowerCase();
}
```
And then use this function in `isString` or `isNumber` functions as below:
```js
function isNumber(value: any): boolean {
    return getType(value) === "number";
}
```

**Review by ali-master — CHANGES_REQUESTED** [review]
Reviewed

**Comment by ali-master** [comment]
Hi @masoudDaliriyan Jan,

Can you complete the PR and make it ready for merge?

**Comment by masoudDaliriyan** [comment]
Hi @ali-master
I'm working on that to night

**Pull request closed without merging**